spring-data-jpa/src/test/java/org/springframework/data/jpa/repository/query/QueryEnhancerUnitTests.java

```diff
@@ -889,6 +890,18 @@ void multipleWithStatementsWorksWithJSQLParser() {
 		assertThat(queryEnhancer.hasConstructorExpression()).isFalse();
 	}
 
+	@Test // GH-2641
+	void mergeStatementWorksWithJSqlParser() {
+		String query = "merge into a using (select id, value from b) query on (a.id = query.id) when matched then update set a.value = value";
+		StringQuery stringQuery = new StringQuery(query, true);
+		QueryEnhancer queryEnhancer = QueryEnhancerFactory.forQuery(stringQuery);
+
+		assertThat(queryEnhancer.getJoinAliases()).isEmpty();
+		assertThat(queryEnhancer.detectAlias()).isNull();
```

Comment:
I don't know if this is ever needed but as far as I can see a merge has an "alias" (see screenshot). 
Maybe it should not return `NULL` here. Is it possible to adapt the `detectAlias()` to consider this edge case without making the method overly complicated? 

![image](https://user-images.githubusercontent.com/25965672/192026675-a46d3065-ac5c-4df8-a7cb-0856b9b50335.png)

Reply:
I tried to adapt the `detectAlias()` method for JSqlParserQueryEnhancer class, however it is more difficult for me to modify the method for the `DefaultQueryEnhancer` ([82590bf](https://github.com/spring-projects/spring-data-jpa/pull/2642/commits/82590bf25b5b6281da6e0844f3957f1ef0c507d5))
I am waiting your feedback